code/modules/mob/living/carbon/life.dm

```diff
@@ -108,15 +109,23 @@
 
 				breath = loc.remove_air(breath_moles)
 		else //Breathe from loc as obj again
+			is_on_internals = TRUE
+
 			if(isobj(loc))
 				var/obj/loc_as_obj = loc
 				loc_as_obj.handle_internal_lifeform(src,0)
 
-	check_breath(breath)
-
+	try_breathing_sound(is_on_internals, breath)
 	if(breath)
 		loc.assume_air(breath)
 
+//Tries to play the carbon a breathing sound when using internals, also invokes check_breath
+/mob/living/carbon/proc/try_breathing_sound(internals, breathing)
+	if(check_breath(breathing) && internals && canon_client?.prefs?.read_preference(/datum/preference/toggle/sound_breathing))
+		breathing_loop.start()
+	else
+		breathing_loop.stop()
```

Comment:
```suggestion
	var/should_be_on = internals && check_breath(breathing) && canon_client?.prefs?.read_preference(/datum/preference/toggle/sound_breathing)
	if(should_be_on && !breathing.timer_id)
		breathing_loop.start()
	else if(!should_be_on && breathing.timer_id)
		breathing_loop.stop()
```
We pretty don't want to keep starting or stopping the loop when there's no need to.

Reply:
I didn't know if it would be a good idea to put another if in there considering this runs for every breathing carbon, but sure.